utils/warnings.ml

```diff
@@ -604,10 +604,13 @@ let message = function
   | Unused_module s -> "unused module " ^ s ^ "."
   | Unboxable_type_in_prim_decl t ->
       Printf.sprintf
-        "This primitive declaration uses type %s, which is unannotated and\n\
-         unboxable. The representation of such types may change in future\n\
-         versions. You should annotate the declaration of %s with [@@boxed]\n\
-         or [@@unboxed]." t t
+        "This primitive declaration uses type %s, whose representation\n\
+         may be either boxed or unboxed -- without an annotation to indicate\n\
+         which representation is intended. The default representation\n\
+         of such types may change in future versions.\n\
+         You should explicitly annotate the declaration of %s\n\
+         with [@@boxed] or [@@unboxed], so that its external interface\n\
+         remains stable in the future." t t
```

Comment:
I am not fond of using the ascii encoding for em dash in text. But it may be a good indication that the grammatical structure is a bit too fancy. Maybe it could be simplified(?) to

```
This primitive declaration uses type %s, whose representation
may be either boxed or unboxed.
Without an annotation to indicate which representation is intended,
the default representation (boxed) has been selected.
This default representation may change in future versions.
...
```
?

Reply:
Thanks, I updated the wording.